backpack-compose/src/main/kotlin/net/skyscanner/backpack/compose/flare/internal/BpkFlareImpl.kt

```diff
@@ -0,0 +1,153 @@
+import net.skyscanner.backpack.compose.flare.BpkFlareRadius
+import net.skyscanner.backpack.compose.tokens.BpkDimension
+
+@Composable
+internal fun FlareContent(
+  pointerDirection: BpkFlarePointerDirection,
+  insetContent: Boolean,
+  contentAlignment: Alignment,
+  content: @Composable BoxScope.() -> Unit,
+) {
+  val contentPadding = when (insetContent) {
+    true -> FlareHeight.dp
+    false -> 0.dp
+  }
+  Box(
+    modifier = Modifier.padding(
+      top = when (pointerDirection) {
+        BpkFlarePointerDirection.Up -> contentPadding
+        BpkFlarePointerDirection.Down -> 0.dp
+      },
+      bottom = when (pointerDirection) {
+        BpkFlarePointerDirection.Up -> 0.dp
+        BpkFlarePointerDirection.Down -> contentPadding
+      }
+    ),
+    propagateMinConstraints = true,
+    contentAlignment = contentAlignment,
+    content = content,
+  )
+}
+
+internal class FlareShape(
+  private val radius: BpkFlareRadius,
+  private val pointerDirection: BpkFlarePointerDirection,
+) : Shape {
+
+  override fun createOutline(size: Size, layoutDirection: LayoutDirection, density: Density): Outline {
+    return Outline.Generic(
+      path = drawFlarePath(size, density.density, radius, pointerDirection)
+    )
+  }
+
+  private fun drawFlarePath(
+    size: Size,
+    density: Float,
+    radius: BpkFlareRadius,
+    pointerDirection: BpkFlarePointerDirection,
+  ): Path {
+    return Path().apply {
+      val flareHeight = FlareHeight * density
+      var scale = flareHeight / FlareVectorHeight
+      if (pointerDirection == BpkFlarePointerDirection.Up) {
+        scale *= -1
+      }
+      addRect(flareHeight, size, density, radius, pointerDirection)
+      addFlare(flareHeight, scale, size, pointerDirection)
+    }
+  }
+}
+
+private fun Path.addRect(
+  flareHeight: Float,
+  size: Size,
+  density: Float,
+  radius: BpkFlareRadius,
+  pointerDirection: BpkFlarePointerDirection,
+) {
+  val borderRect = when (pointerDirection) {
+    BpkFlarePointerDirection.Up -> Rect(0f, flareHeight, size.width, size.height)
+    BpkFlarePointerDirection.Down -> Rect(0f, 0f, size.width, size.height - flareHeight)
+  }
+  when (radius) {
+    BpkFlareRadius.None -> addRect(borderRect)
+    BpkFlareRadius.Medium -> {
+      val cornerRadius = BpkDimension.BorderRadius.Md.value * density
+      addRoundRect(RoundRect(borderRect, CornerRadius(cornerRadius, cornerRadius)))
+    }
+  }
+}
+
+private fun Path.addFlare(
+  flareHeight: Float,
+  scale: Float,
+  size: Size,
+  pointerDirection: BpkFlarePointerDirection,
+) {
+  val flareWidth = scale * FlareVectorWidth
+  val flareX = (size.width - flareWidth) / 2
+  val flareY = when (pointerDirection) {
+    BpkFlarePointerDirection.Up -> flareHeight
+    BpkFlarePointerDirection.Down -> size.height - flareHeight
+  }
+  moveTo(flareX + 4.329f * scale, flareY)
+  cubicTo(
+    flareX + 14.663905f * scale, flareY + 0.409248008f * scale,
```

Comment:
How did we get these constants?

Reply:
And how iOS get it? :)

Reply:
That code long outlives the current set of iOS engineers, but my assumption would be they got an svg and mapped the svg to code. I tried to do that, but the flare isn't exportable from Figma for some reason. I also tried to convert the android vector to a compose vector and use that, but that contains a lot of relative position changes which aren't available in a simple Path, so this was the simpler approach and seems to match the current Flare

Reply:
It may be also worth exposing it to SVG -> trying to use Vector2Compose to get it done. These shapes are in general generated by such tools.

Otherwise, it looks good, I'd just inline drawFlarePath() method. Also, please consider using 4 different shape object instead of using a new one.

Reply:
It just describes the path and does not draw anything. Why do you think inlining this method decreases the performance?

Reply:
FlareBottom, FlatBottomMediumRadius, FlatTop, FlareTopMediumRadius

Reply:
Not decreasing, but also not improving. It results in a warning
<img width="768" alt="Screenshot 2022-06-08 at 16 31 09" src="https://user-images.githubusercontent.com/1529791/172657062-680827b2-0b4d-4fce-b3eb-c7404e441080.png">